src/prompts/checkly.ts

```diff
@@ -46,6 +51,207 @@ ${stringify(entry)}`;
+
+  // Place checks into appropriate clusters based on time and location
+  checkResults.forEach((check) => {
+    const checkTime = new Date(check.startedAt).getTime();
+    const timeIndex = Math.floor(
+      (checkTime - startDate.getTime()) /
+        (1000 * 60 * checkDetails.frequency * clusterFactor),
+    );
+    const locationIndex = checkDetails.locations.indexOf(check.location);
+
+    if (timeIndex >= 0 && timeIndex < timeClusters && locationIndex >= 0) {
+      clusters[timeIndex][locationIndex] = [
+        ...clusters[timeIndex][locationIndex],
+        check.id,
+      ];
+    }
+  });
+
+  // Filter and flatten clusters into one-dimensional array with metadata
+  const filteredClusters = clusters
+    .flatMap((timeSlice, timeIndex) =>
+      timeSlice.map((locationChecks, locationIndex) => ({
+        timeIndex,
+        location: checkDetails.locations[locationIndex],
+        checks: locationChecks,
+      })),
+    )
+    .filter((cluster) => cluster.checks.length > 0)
+    .map((cluster) => {
+      const clusterStartTime = new Date(
+        startDate.getTime() +
+          cluster.timeIndex *
+            1000 *
+            60 *
+            checkDetails.frequency *
+            clusterFactor,
+      );
+      const clusterEndTime = new Date(
+        clusterStartTime.getTime() +
+          1000 * 60 * checkDetails.frequency * clusterFactor,
+      );
+
+      return {
+        location: cluster.location,
+        checks: cluster.checks,
+        startTime: clusterStartTime.toISOString(),
+        endTime: clusterEndTime.toISOString(),
+      };
+    });
+
+  return filteredClusters;
+}
+
+export function summarizeTestStepsPrompt(
+  testName: string,
+  scriptName: string,
+  scriptPath: string,
+  dependencies: { script: string; scriptPath: string }[],
+  errors: string[],
+): [string, PromptConfig] {
+  return [
+    `
+      The following details describe a test which is used to monitor an application.
+
+      Do not refer to technical details of the test, use the domain language from the application under test.
+      Test name: ${testName}
+      Script name: ${scriptName}
+      Script: ${scriptPath}
+      Error stack: ${errors}
+
+      Dependent scripts of the main script:
+      ${dependencies.map((d) => `- ${d.scriptPath}\n  ${d.script}`).join("\n")}
+
+      Summarize the steps executed by the test using high level domain language. Focus on the user flow omit technical details. Use max 5 words per step.
+      Identify the step which failed by match the script code with the stack of the error. Include details about the test failure.
+
+      CONSTITUTION:
+      - Always prioritize accuracy and relevance in the summary
+      - Be concise but comprehensive in your explanations
+      - Focus on providing actionable information that can help judging user impact
+    `,
+    promptConfig("checklySummarizeFeatureCoverage", {
+      temperature: 1,
```

Comment:
just curious, why are you using temperature 1 here, and 0.1 for summarising errors?